### ISSUE TITLE
Error reported for explicit conversion from native integer to enum

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#48035]
Explicit conversions should be allowed from native integers to enum values.
```C#
enum E { }

class Program
{
    static void Main()
    {
        nint i = 1;
        nuint u = 2;
        E e;
        e = (E)i;
        e = (E)u;
    }
}
```
Expected: No errors

Result:
```
(10,13): error CS0030: Cannot convert type 'nint' to 'E'
(11,13): error CS0030: Cannot convert type 'nuint' to 'E'
```

**Comment by jcouv** [comment]
Do we have a workaround for `runtime` folks, or do we plan to address this in 16.8?

**Comment by cston** [comment]
A workaround is to cast to `int` or `uint` first:
```C#
e = (E)(int)i;
e = (E)(uint)u;
```

**Comment by Youssef1313** [comment]
I think the bug is in the following method:

```csharp
        public static bool IsNumericType(SpecialType specialType)
        {
            switch (specialType)
            {
                case SpecialType.System_Char:
                case SpecialType.System_SByte:
                case SpecialType.System_Byte:
                case SpecialType.System_Int16:
                case SpecialType.System_UInt16:
                case SpecialType.System_Int32:
                case SpecialType.System_UInt32:
                case SpecialType.System_Int64:
                case SpecialType.System_UInt64:
                case SpecialType.System_Single:
                case SpecialType.System_Double:
                case SpecialType.System_Decimal:
                    return true;
                default:
                    return false;
            }
        }
```

I'll open a PR to address that.

**Comment by cston** [comment]
Thanks @Youssef1313. I'm already working on a fix.

**Comment by Youssef1313** [comment]
Good to know!

Consider also the implicit enumeration conversion case:

```csharp
enum E { }

class Program
{
    static void Main()
    {
        const nint x = 0;
        const int y = 0;
        E e;
        e = x; // unexpected compile error.
        e = y; // compiles as expected.
    }
}
```
